Makefile

```diff
@@ -1,6 +1,127 @@
+GOOS ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
+ARCH ?= $(shell uname -m)
+TARGET_IP ?= 127.0.0.1
+API_LEVEL ?= 29
+MOD_VERSION ?= 0.0.1
 
-bin/viamrtsp: *.go cmd/module/*.go
-	go build -o bin/viamrtsp cmd/module/cmd.go
+UNAME=$(shell uname)
+ifeq ($(UNAME),Linux)
+	NDK_ROOT ?= $(HOME)/Android/Sdk/ndk/26.1.10909125
+	HOST_OS ?= linux
+	CC_ARCH ?= aarch64
+else
+	NDK_ROOT ?= $(HOME)/Library/Android/sdk/ndk/26.1.10909125
+	HOST_OS ?= darwin
+	CC_ARCH ?= aarch64
+endif
+
+# FFmpeg build settings
+TOOLCHAIN := $(NDK_ROOT)/toolchains/llvm/prebuilt/$(HOST_OS)-x86_64
+CC := $(TOOLCHAIN)/bin/$(CC_ARCH)-linux-android$(API_LEVEL)-clang
+CXX := $(TOOLCHAIN)/bin/$(CC_ARCH)-linux-android$(API_LEVEL)-clang++
+AR := $(TOOLCHAIN)/bin/llvm-ar
+LD := $(CC)
+RANLIB := $(TOOLCHAIN)/bin/llvm-ranlib
+STRIP := $(TOOLCHAIN)/bin/llvm-strip
+NM := $(TOOLCHAIN)/bin/llvm-nm
+SYSROOT := $(TOOLCHAIN)/sysroot
+
+FFMPEG_SUBDIR=viamrtsp/ffmpeg-android
+FFMPEG_PREFIX=$(HOME)/$(FFMPEG_SUBDIR)
+
+# CGO settings
+CGO_ENABLED := 1
+CGO_CFLAGS := -I$(FFMPEG_PREFIX)/include
+CGO_LDFLAGS := -L$(FFMPEG_PREFIX)/lib
+
+# Output settings
+OUTPUT_DIR := bin
+OUTPUT := $(OUTPUT_DIR)/viamrtsp-$(GOOS)-$(GOARCH)
+APPIMG := rtsp-module-$(MOD_VERSION)-$(ARCH).AppImage
```

Comment:
do we really need appimage? 
really trying to move away from it.

Reply:
Good call! We do not need an AppImage. 
- Tested that packaging FFmpeg libs, go binary, and shell entry works for Linux.
- Going to try out a static FFmpeg build/link so that we can just ship the go binary.